src/lib.rs

```diff
@@ -101,7 +101,7 @@ impl ToKey for String {
 
 pub fn encode<P: ToKey>(mut header: JsonValue, signing_key: &P, payload: &JsonValue, algorithm: Algorithm) -> Result<String, Error> {
     header["alg"] = JsonValue::String(algorithm.to_string());
-    header["typ"] = JsonValue::String(STANDARD_HEADER_TYPE.to_owned());
+    if !header["typ"].is_string() {header["typ"] = JsonValue::String(STANDARD_HEADER_TYPE.to_owned());}
```

Comment:
Style: Don't do one-line `if` statements - split after the opening brace.  I would suggest running `cargo fmt`, but that may well reformat other parts of the code, which you don't want to do.

I think the check might be better testing against None:
```
if header["typ"] == None {
    header["typ"] = JsonValue::String(STANDARD_HEADER_TYPE.to_owned());
}
```

Reply:
I initially tried header["typ"] ==None, but rust states: can't compare `serde_json::Value` with `std::option::Option<_>`. 
There is a .is_null() that works the same way but that does create a regression where header["typ"] is set to a non-string value by mistake

Reply:
Sorry, the value returned if the key does not exist is `Value::Null`: https://github.com/serde-rs/json

Yeah, you need to decide what the behavior should be if the user passes in a non-string.  I thought it would be better to allow it through and then get an error returned later on when it tries to use the headers (which I presume will give a reasonable error - you may want to check).  With your current implementation you will silently replace the user's supplied non-string value, which they won't be expecting and almost certainly won't be what they want!